File: src/main/java/com/example/camunda/delegate/CommonDelegate.java
package com.example.camunda.delegate;

import org.camunda.bpm.engine.delegate.DelegateExecution;
import org.camunda.bpm.engine.delegate.JavaDelegate;

import java.lang.reflect.Method;

public abstract class CommonDelegate implements JavaDelegate {
    @Override
    public void execute(DelegateExecution execution) throws Exception {
        String currentActivityName = execution.getCurrentActivityName();
        String methodName = currentActivityName;
        Method declaredMethod = getClass().getDeclaredMethod(methodName, DelegateExecution.class);
        declaredMethod.invoke(execution);
    }
}

File: backend/src/main/kotlin/com/ptah/camunda/delegate/CommonDelegate.kt
package com.ptah.camunda.delegate

import org.camunda.bpm.engine.delegate.DelegateExecution
import org.camunda.bpm.engine.delegate.JavaDelegate

abstract class CommonDelegate : JavaDelegate {
    @Throws(Exception::class)
    override fun execute(execution: DelegateExecution) {
        val currentActivityName = execution.currentActivityName
        val declaredMethod = javaClass.getDeclaredMethod(currentActivityName, DelegateExecution::class.java)
        declaredMethod.invoke(execution)
    }
}
